src/config/mod.rs

```diff
@@ -872,6 +879,28 @@ impl SetOpt for EnableMetrics {
     }
 }
 
+/// The ip protocol version to use
+#[derive(Clone, Debug)]
+pub enum IpProtocol {
```

Comment:
Bikeshedding: Since IP stands for Internet Protocol, this enum name seems a little redundant. (Internet Protocol Protocol?) How about `IpVersion`?

Ditto for `Configurable::ip_protocol`.